src/Storages/StorageS3.cpp

```diff
@@ -503,6 +569,13 @@ Chunk StorageS3Source::generate()
         {
             UInt64 num_rows = chunk.getNumRows();
 
+            auto it = object_infos.find(file_path);
+            if (num_rows && it != object_infos.end())
+            {
+                updateRowsProgressApprox(
+                    *this, chunk, file_iterator->getTotalSize(), total_rows_approx_accumulated, total_rows_count_times, total_rows_approx_max);
```

Comment:
In case of large number of files they may be listed with several requests in `DisclosedGlobIterator` (by default batch size is 1000), so `getTotalSize()` may return incomplete number of result rows. But probably it's ok, since it will work in most of cases and I don't know better solution.